### PR TITLE
Improve the documentation regarding JOIN support in continuous aggregates

### PR DESCRIPTION
**Pull request by RafiaSabih**
With Timescledb v 2.10, JOINS are supported in continuous aggregates. The important caveats to this are:
1. JOIN has to be between only one hypertable (and NOT a continuous aggregate) and one normal PostgreSQL table
2. Joins type to be INNER.
3. Join condition can only be equality.
4. Supoort of USING clause in PostgreSQL v12 is not permitted, but joins can still be used with JOINS clause.

# Description

[Short summary of why you created this PR]

# Links

Fixes #[insert issue link, if any]

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs/)

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

### CODE REVIEW

File: api/create_materialized_view.md
Comment (iroussos) [inline api/create_materialized_view.md:53]: @RafiaSabih can you please check and point out any issues with my proposed update? I also added that we don't track the regular PG table for invalidations

```suggestion
1. Only joins between two tables are currently supported, where one is a hypertable and the other table is a normal PostgreSQL table. The order of tables in the `JOIN` clause does not matter.
2. Only the `INNER` `JOIN` type is currently supported.
3. Only equality conditions are currently supported as `JOIN` conditions.
4. Joins in hierarchical continuous aggregates are not currently supported. The hypertable on the `JOIN` condition must be a normal hypertable, not a continuous aggregate itself.
5. Changes on the normal PostgreSQL table are not tracked. Only changes on the hypertable result in updates to the continuous aggregate when it is refreshed.
```

File: api/create_materialized_view.md
Comment (iroussos) [inline api/create_materialized_view.md:51]: Should we call those `normal` or `regular` tables?

What's your opinion @Loquacity @RafiaSabih?

File: timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md
Comment (iroussos) [inline timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md:169]: We should at least update this with whatever restrictions we add above.

But this also worries me because we have the same things twice. I would prefer if we have everything in one place and then link to that canonical section for joins on caggs from everywhere.

**Review by iroussos — COMMENTED** [review]
Thank you @RafiaSabih,

I have added a few comments for you to check and then let's send it to @Loquacity for her review.

@Loquacity as this blocks announcing 2.10.0, I propose that we do an initial iteration and merge in whatever form/way you prefer to have the proposed updates and comments. We can then link to whatever page/section we'll have as a SSOT, get the announcement out and revisit afterwards

File: api/create_materialized_view.md
Comment (RafiaSabih) [reply api/create_materialized_view.md:51]: either is fine by me, no preferences there.

Comment (Loquacity) [reply api/create_materialized_view.md]: We use "standard PostgreSQL": https://docs.timescale.com/timescaledb/latest/contribute-to-docs/#v

File: api/create_materialized_view.md
Comment (iroussos) [inline api/create_materialized_view.md:52]:
```suggestion
1. Only joins between two tables are currently supported, where one is a hypertable and the other table is a standard PostgreSQL table. The order of tables in the `JOIN` clause does not matter.
```

File: api/create_materialized_view.md
Comment (iroussos) [inline api/create_materialized_view.md:57]:
```suggestion
6. Support for `USING` in the `JOIN` clause requires PostgreSQL&nbsp;13 or later. PostgreSQL&nbsp;12 is supported for joins with a condition in the `WHERE` or `JOIN` clauses.

```

**Review by iroussos — APPROVED** [review]

**Comment by iroussos** [comment]
@Loquacity yours for a final check and merge; Feel free to do whatever changes you think are necessary, but let's merge this iteration without further back and forth as it is blocking the next steps for 2.10.0. Thank you!

File: timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md
Comment (Loquacity) [inline timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md:164]:
```suggestion
For more information, including some additional restrictions, see the 
[continuous aggregates section][api-cagg-create_materialized_view]. 
```

File: timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md
Comment (iroussos) [reply timescaledb/how-to-guides/continuous-aggregates/create-a-continuous-aggregate.md]: Let's merge this version as it is blocking the release announcement and follow-up on that with a new PR

File: api/create_materialized_view.md
Comment (iroussos) [inline api/create_materialized_view.md:57]:
```suggestion
In TimescaleDB&nbsp;2.10.0 and later, the `FROM` clause supports `JOINS`, with these restrictions:
*   To join between two tables, one table must be a hypertable and the other table must 
     be a standard PostgreSQL table. The order of tables in the `JOIN` clause does not 
     matter.
*   You must use an `INNER JOIN`, no other join type is supported.
*   The `JOIN` conditions can only be equality conditions.
*   The hypertable on the `JOIN` condition must be a hypertable, and not a continuous 
     aggregate. Additionally, you can't use joins in hierarchical continuous aggregates. 
*   Changes to the hypertable are tracked, and are updated in the continuous aggregate 
     when it is refreshed. Changes to the standard PostgreSQL table are not tracked. 
*   The `USING` clause is supported in joins for PostgreSQL&nbsp;13 or later.
     In PostgreSQL&nbsp;12 only joins with a condition in the `WHERE` clause or `JOIN` clause can be used.
```

**Comment by iroussos** [comment]
Thank you @Loquacity, I have applied your suggestions; looks much better now!

Feel free to make any additional changes you deem as necessary if something feels wrong without a further review from me and merge.

**Review by Loquacity — APPROVED** [review]

**Pull request merged**
